File: mobile/lib/models/equipment.dart
class Equipment {
  final int id;
  final String name;
  final String? reference;
  final String qrToken;
  final String? category;
  final String? location;
  final String status;
  final String? manufacturer;
  final String? model;
  final String? serialNumber;
  final String? notes;

  Equipment({
    required this.id,
    required this.name,
    this.reference,
    required this.qrToken,
    this.category,
    this.location,
    required this.status,
    this.manufacturer,
    this.model,
    this.serialNumber,
    this.notes,
  });

  factory Equipment.fromJson(Map<String, dynamic> json) {
    return Equipment(
      id: json['id'],
      name: json['name'],
      reference: json['reference'],
      qrToken: json['qr_token'],
      category: json['category'],
      location: json['location'],
      status: json['status'],
      manufacturer: json['manufacturer'],
      model: json['model'],
      serialNumber: json['serial_number'],
      notes: json['notes'],
    );
  }

  String get statusLabel => switch (status) {
        'operational' => 'Opérationnel',
        'down' => 'En panne',
        'maintenance' => 'En maintenance',
        _ => status,
      };
}

File: backend/app/Models/Equipment.php
<?php

namespace App\Models;

use App\Models\Concerns\BelongsToTenant;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Str;

class Equipment extends Model
{
    use BelongsToTenant, HasFactory;

    protected $table = 'equipments';

    protected $fillable = [
        'tenant_id',
        'name',
        'reference',
        'qr_token',
        'category',
        'location',
        'status',
        'manufacturer',
        'model',
        'serial_number',
        'purchase_date',
        'photo_path',
        'notes',
    ];

    protected $casts = [
        'purchase_date' => 'date',
    ];

    protected static function booted(): void
    {
        static::creating(function (Equipment $equipment) {
            if (empty($equipment->qr_token)) {
                $equipment->qr_token = (string) Str::uuid();
            }
        });
    }

    public function tickets()
    {
        return $this->hasMany(Ticket::class);
    }

    public function preventivePlans()
    {
        return $this->hasMany(PreventivePlan::class);
    }

    public function workOrders()
    {
        return $this->hasMany(WorkOrder::class);
    }

    public function interventions()
    {
        return $this->hasMany(Intervention::class);
    }
}
